src/ngx_http_lua_semaphore.c

```diff
@@ -557,8 +557,9 @@ ngx_http_lua_ffi_sema_gc(ngx_http_lua_sema_t *sem)
         return;
     }
 
-    if (!ngx_queue_empty(&sem->wait_queue)) {
-        ngx_log_error(NGX_LOG_CRIT, ngx_cycle->log, 0,
+    if (!ngx_terminate && !ngx_quit
```

Comment:
@membphis OK. Maybe we also need something similar for `lua_code_cache off`. In that case, the Lua VM gets destroyed when the request handler is done.